File: food-order/app/domain/order.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Order struct {
	Id          int
	Uuid        uuid.UUID
    Number      int
	Status      OrderStatus
    UserUuid    uuid.UUID
	Restaurant  Restaurant
    Payment     PaymentOrder
	Items       []OrderItem
	Total       float64
	DateCreated time.Time
	DateUpdated time.Time
}

type OrderItem struct {
	Id           int
	Uuid         uuid.UUID
	MenuItemUuid uuid.UUID
	Name         string
	Amount       int
	UnitValue    float64
}

File: food-order/Domain/Order.cs
using System.Collections.Generic;

namespace food_order.Domain
{
    public class Order
    {
        public long? Id { get; }
        public string Uuid { get; }
        public Restaurant.Restaurant Restaurant { get; }
        public List<OrderItem> Itens { get; }
        public decimal Total { get; }

        public Order(long? id, string uuid, Restaurant.Restaurant restaurant, List<OrderItem> itens, decimal total) 
        {
            this.Id = id;
            this.Uuid = uuid;
            this.Restaurant = restaurant;
            this.Itens = itens;
            this.Total = total;
        }
    }
}
